src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/EditionService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions.Collections.Facades.Differential;
+using Uno.Extensions.Reactive.Dispatching;
+using Uno.Extensions.Threading;
+
+namespace Uno.Extensions.Reactive.Bindings.Collections.Services;
+
+internal class EditionService : IEditionService, IDisposable
+{
+	private readonly FastAsyncLock _gate = new();
+	private readonly CancellationTokenSource _ct = new();
+	private readonly AsyncAction<Func<IDifferentialCollectionNode, IDifferentialCollectionNode>> _editFromView;
+
+	private ImmutableQueue<Func<IDifferentialCollectionNode, IDifferentialCollectionNode>> _editions = ImmutableQueue<Func<IDifferentialCollectionNode, IDifferentialCollectionNode>>.Empty;
+
+	public EditionService(AsyncAction<Func<IDifferentialCollectionNode, IDifferentialCollectionNode>> editFromView)
+	{
+		_editFromView = editFromView;
+	}
+
+	/// <inheritdoc />
+	public void Update(Func<IDifferentialCollectionNode, IDifferentialCollectionNode> edition)
+	{
+		ImmutableInterlocked.Enqueue(ref _editions, edition);
+		if(DispatcherHelper.GetForCurrentThread() is {} dispatcher)
+		{
+			Task.Run(
```

Comment:
Maybe not relevant, but is ordering a problem here? Inner execution may not be serialized according to `Update` invocations.

Reply:
No the thread safety / sequencing is ensured by the `ImmutableInterlocked.Enqueue`. Then we schedule a `Dequeue` on another thread which will execute what is available so far.